src/ucp/core/ucp_ep.c

```diff
@@ -3568,28 +3568,34 @@ ucs_status_t ucp_ep_query(ucp_ep_h ep, ucp_ep_attr_t *attr)
     return UCS_OK;
 }
 
-ucs_status_t ucp_ep_realloc_slow_lanes(ucp_ep_h ep, uint32_t old_num_lanes, uint32_t new_num_lanes)
+ucs_status_t ucp_ep_realloc_slow_lanes(ucp_ep_h ep, unsigned new_num_lanes)
 {
+    int num_slow_lanes   = new_num_lanes - UCP_MAX_FAST_PATH_LANES;
+    ucp_ep_ext_t *ep_ext = ep->ext;
     ucp_lane_index_t lane;
-    int num_slow_lanes;
+    unsigned old_num_lanes;
+    uct_ep_h *tmp;
 
-    if (new_num_lanes <= UCP_MAX_FAST_PATH_LANES) {
-        ucs_free(ep->ext->uct_eps);
-        ep->ext->uct_eps = NULL;
+    if (num_slow_lanes <= 0) {
+        ucs_free(ep_ext->uct_eps);
+        ep_ext->uct_eps = NULL;
         return UCS_OK;
     }
 
-    num_slow_lanes = new_num_lanes - UCP_MAX_FAST_PATH_LANES;
-
-    ep->ext->uct_eps = ucs_realloc(ep->ext->uct_eps,
-                                   num_slow_lanes * sizeof(uct_ep_h),
-                                   "ext_uct_eps");
-    if (ep->ext->uct_eps == NULL) {
+    tmp = ucs_realloc(ep_ext->uct_eps,
+                      num_slow_lanes * sizeof(*ep_ext->uct_eps),
+                      "ucp_slow_lanes");
+    if (tmp == NULL) {
         return UCS_ERR_NO_MEMORY;
     }
 
-    for (lane = old_num_lanes; lane < new_num_lanes;
-         ++lane) {
+    ep_ext->uct_eps = tmp;
+
+    old_num_lanes = (ep->cfg_index != UCP_WORKER_CFG_INDEX_NULL) ?
+                            ucp_ep_num_lanes(ep) :
+                            UCP_MAX_FAST_PATH_LANES;
```

Comment:
shouldn't it be 0?

Reply:
we already fill the fast lanes with NULL in `ucp_ep_allocate`

Reply:
IMO it's more consistent we consider UCP_WORKER_CFG_INDEX_NULL as 0 lanes, even though we set these lanes to NULL beforehand